Standards/scs-01XX-v1-secure-connections.md

```diff
@@ -0,0 +1,178 @@
+  - If using RabbitMQ, the OpenStack services' oslo.messaging configuration for RabbitMQ MUST use the `kombu_ssl_*` options accordingly to enable SSL.
+  - If using Qpid, the OpenStack services' oslo.messaging configuration for Qpid MUST set the `qpid_protocol` option to `ssl` to enable SSL.
+- If using Apache Kafka, the server listener MUST be configured to accept SSL connections. See [Apache Kafka Listener Configuration](https://kafka.apache.org/documentation/#listener_configuration).
+  - The OpenStack services' oslo.messaging configuration for Kafka MUST specify `security_protocol` as either `SSL` or `SASL_SSL` and the related options appropriately. See [Kafka Driver Options](https://docs.openstack.org/oslo.messaging/latest/admin/kafka.html#driver-options).
+
+### Hypervisor and Live Migration Connections
+
+- If using libvirt on compute nodes, the libvirt port (as per `listen_addr`) MUST NOT be exposed to the network in an unauthenticated and unprotected fashion. It SHOULD be bound to `127.0.0.1`.
+- If QEMU and libvirt are used as the hypervisor interface in Nova, QEMU-native TLS SHOULD be used. See [Secure live migration with QEMU-native TLS](https://docs.openstack.org/nova/latest/admin/secure-live-migration-with-qemu-native-tls.html).
+
+### External VM Connections
+
+- As an OPTIONAL measure to assist customers in protecting external connections to their OpenStack networks and VMs, the infrastructure MAY provide VPNaaS solutions to users.
+  - For example the Neutron VPNaaS service MAY be integrated into the infrastructure with the Neutron VPNaaS API extension enabled. See the [Neutron VPNaaS documentation](https://docs.openstack.org/neutron-vpnaas/latest/).
+
+### Internal Neutron Connections
+
+- As an OPTIONAL measure to protect Neutron SDN traffic between physical nodes within the infrastructure, IPsec tunnels MAY be established between nodes involved in Neutron networks, such as compute and network controller nodes, at the network interfaces configured in Neutron.
+
+## Related Documents
+
+- [OpenStack Security Guide](https://docs.openstack.org/security-guide/)
+  - [OpenStack Security Guide: Secure communication](https://docs.openstack.org/security-guide/secure-communication.html)
+  - [OpenStack Security Guide: Database transport security](https://docs.openstack.org/security-guide/databases/database-transport-security.html)
+  - [OpenStack Security Guide: Messaging transport security](https://docs.openstack.org/security-guide/messaging/security.html#messaging-transport-security)
+- [Nova Documentation: Secure live migration with QEMU-native TLS](https://docs.openstack.org/nova/latest/admin/secure-live-migration-with-qemu-native-tls.html)
+- [Guide to TLS Standards Compliance](https://www.ssl.com/guide/tls-standards-compliance/)
+
+## Conformance Tests
+
+Conformance tests are limited to communication channels exposed to users, such as the public API interfaces.
```

Comment:
Why? At least there should be some justification either here or in a decision record document, no?

Reply:
The automatic conformance tests as I understand them are executed on SCS infrastructure running again remotely reachable cloud infrastructure. Is this correct? This would simply not allow to look into internal communication channels automatically. E.g. the connection from Nova to the DB can only be checked when checking the config files. Same goes for the Message Queue. This can only be done if you have access on the nodes and not just to the OpenStack Infrastructure.

Reply:
While I understand that, I find it difficult to standardize something when the conformance tests for the standard can not really guarantee that the standard is being correctly implemented in it's full form. Imho this leads to a false sense of security.

At least it must be made clear to end users what is not tested (e.g. live migration traffic), and/or maybe the CSP must somehow declare that they in fact follow the standard to the letter, but I find this a much weaker sign than the result of an automated test.

Maybe create an automated test that only the CSP can run internally and mandate that results are published? You still have to trust the CSP to not tamper with the test, but at least the output can be parsed automatically via machines and be made available on scs.community this way.

Reply:
I think this can be marked as resolved. At least from my understanding it's clear that certain things can only be assured/tested by the CSP and not by the end user.

Feel free to unresolve if further work in this area is neeeded.